dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -297,15 +297,11 @@ def _generate_counterfactuals(self, query_instance, total_CFs, initialization="k
 
     def predict_fn_scores(self, input_instance):
         """returns predictions"""
-        input_instance = self.label_decode(input_instance)
-        return self.model.get_output(input_instance)
+        return self.predict_fn(input_instance)
```

Comment:
There's a tricky thing because of how `self.model.get_output()` operates. If model is a classifier, then it returns `predict_proba`. So we need the scores in this function, but in the `predict_fn` for genetic, we need the actual class label. So the previous code uses 'self.model.model.predict'. 

A better solution will be resolve it in the Model class. `self.model.get_output` should always return the scores, and perhaps `get_prediction` should predict the class label. 

Both functions are identical for regression. Paging @soundarya98---what do you think?

Reply:
There are multiple predict functions in dice_genetic and dice_KD:

**1. dice_genetic:**
	- predict_fn_scores: This function first label decodes the input and then feeds the input to the ML classifier (which takes only numbers as input since we moved all transformations to the pipeline).
	If classifier: The return type is the scores (predict_proba output) of the function
	If regressor: The return type is a single 'predict' value
	This function is needed for compute_y_loss
	
        - predict_fn: This function also label decodes the input
           The return type is always a single predict value regardless of whether it is a classifier or regressor
           This function is needed for everything else

**2. dice_KD:**
       - predict_fn: This function always returns the single predict value regardless of whether it is a classifier or   regressor
   
A single function in Base Model with an additional parameter to return scores makes sense and will be clean as well!

Reply:
Like your idea @gaugup. We can define: `get_output(model_score=False)`? I'm thinking that `model_score` may be more intuitive than `actual_predictions`. What do you think?

Reply:
@amit-sharma, yes made the changes to reflect model_score. 

@amit-sharma / @soundarya98 could you help review this. I made the requested changes.

Reply:
Thanks! Overall it looks good. I added a comment on one change in explainerbase for `predict_fn_for_sparsity`.

Reply:
Thank you! The PR looks good to me, I've added two small comments on `get_num_output_nodes2` and the default parameter value for the `get_output` function

Reply:
@amit-sharma and @soundarya98 I responded to your other comments in this PR. Please have a look again tomorrow.